### DISCUSSION TITLE
Formating the displayed search results

### DISCUSSION THREAD

**Discussion by JStyle21** [osm-search/Nominatim#2584] [Q&A] [answered]
Hi,

Currently when searching for something i get a JSON response that contains a display name that holds the full address info i.e. street, city, state, country, district, zip code etc.

Is it somehow possible to change the format? something like street, city, country is much cleaner.
Unfortunately the display variable already comes as a one liner and not an array so it's hard to separate easily and i haven't found anything in the docs that gives me something else i can control.

**Comment by lonvia** [comment, marked as answer]
Try with [GeocodeJSON](https://nominatim.org/release-docs/latest/api/Output/#geocodejson) format. This gives you separate fields for housenumber, street, city, etc. which you can format to your liking.

e.g https://nominatim.openstreetmap.org/search.php?q=berlin&format=geocodejson

**Reply by JStyle21** [reply]
Maybe i'm missing something here because i don't see the fields you mentioned even with geocodejson the label is still 1 field.
If i split it by comma there is no way of knowing which parts to drop since some places have states and districts and other don't.

**Reply by mtmail** [reply]
Add &addressdetails=1 Example https://nominatim.openstreetmap.org/search.php?q=berlin&format=geocodejson&addressdetails=1 or https://nominatim.openstreetmap.org/search.php?q=berlin&format=jsonv2&addressdetails=1

**Reply by JStyle21** [reply]
Thanks that works, mostly (sometimes city is town and vice versa).
Any way to filter by address type? just looking for residential locations at this time.

**Reply by mtmail** [reply]
There's no filter for address type.

**Reply by lonvia** [reply]
If you use the geocodejson format, then the names are consistent. It is just for json and jsonv2 format that you get a lot more varied address types, see https://nominatim.org/release-docs/latest/api/Output/#addressdetails .

**Reply by JStyle21** [reply]
Thank you!
I think that solved all of my issues :+1:
